informer/src/handler/subscription.rs

```diff
@@ -14,28 +14,38 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{ColdEvents, EventTags, Events, Params, Sink};
+use crate::{ColdEvents, EventTags, Events, Params, Sink, SubscriptionId};
 use jsonrpc_core::futures::Future;
 
 #[derive(Clone)]
 enum ConnectionState {
     Connected,
 }
 
+#[derive(Clone)]
+pub enum ColdState {
```

Comment:
The ColdState is checking whether the subscription has a cold event subscription or not. This enum has 3 different states:
- Has: means it has a cold event subscription.
- NotApplicable: means it does not have a cold event subscription. The subscription type is Hot.
- Removed: This means the subscription is removed. 
I used an enum instead of a boolean flag to organize the state.